flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/functions/ScalarFunctions.scala

```diff
@@ -219,6 +220,31 @@ object ScalarFunctions {
     str.replaceAll(regex, Matcher.quoteReplacement(replacement))
   }
 
+  /**
+    * Returns a string extracted with a specified regular expression and a regex match group index.
+    */
+  def regexp_extract(str: String, regex: String, extractIndex: Integer): String = {
+    if (str == null || regex == null) {
+      return null
+    }
+
+    val m = Pattern.compile(regex).matcher(str)
+    if (m.find) {
+      val mr = m.toMatchResult
+      return mr.group(extractIndex)
+    }
+
+    ""
```

Comment:
I think when no matching found, the method should return null instead of an empty string, right?

Reply:
yes, will fix it, hold on.y

Reply:
Done.